indiserver/DvrInfo.cpp

```diff
@@ -0,0 +1,325 @@
+/* INDI Server for protocol version 1.7.
+ * Copyright (C) 2007 Elwood C. Downey <ecdowney@clearskyinstitute.com>
+                 2013 Jasem Mutlaq <mutlaqja@ikarustech.com>
+                 2022 Ludovic Pollet
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+#include "DvrInfo.hpp"
+#include "Utils.hpp"
+#include "Constants.hpp"
+#include "Msg.hpp"
+#include "ClInfo.hpp"
+#include "Property.hpp"
+
+ConcurrentSet<DvrInfo> DvrInfo::drivers;
+
+void DvrInfo::onMessage(XMLEle * root, std::list<int> &sharedBuffers)
+{
+    char *roottag    = tagXMLEle(root);
+    const char *dev  = findXMLAttValu(root, "device");
+    const char *name = findXMLAttValu(root, "name");
+    int isblob       = !strcmp(tagXMLEle(root), "setBLOBVector");
+
+    if (verbose > 2)
+        traceMsg("read ", root);
+    else if (verbose > 1)
+    {
+        log(fmt("read <%s device='%s' name='%s'>\n",
+                tagXMLEle(root), findXMLAttValu(root, "device"), findXMLAttValu(root, "name")));
+    }
+
+    /* that's all if driver is just registering a snoop */
+    /* JM 2016-05-18: Send getProperties to upstream chained servers as well.*/
+    if (!strcmp(roottag, "getProperties"))
+    {
+        this->addSDevice(dev, name);
```

Comment:
`dev` is not checked for nullness before this is used. Again `basic_string` cannot be constructed from nullptr and this will crash.